flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -235,11 +230,44 @@ public void run(SourceContext<OUT> ctx) throws Exception {
 					sessionIds.add(deliveryTag);
 				}
 
-				ctx.collect(result);
+				schema.deserialize(delivery.getBody(), collector);
```

Comment:
Is it ok that we now only "break" out of the loop after the `autoAck` handling?

Reply:
It is the same behaviour change as we did in Kafka. That we "try" to checkpoint the last element. Before as you are saying it was not checkpointed. Still the chances that it will actually be checkpointed are rather slim. Commiting of this acks happen on checkpoint. So the checkpoint would've to be triggered between breaking out of the loop and finishing the source task.

That said I think there should be no to minor implications of this change.

Reply:
Ok, thanks for the reminder!